Cpp_test/lib/ipc-shm-receive.cpp

```diff
@@ -0,0 +1,110 @@
+#include "ipc-shm-receive.h"
+
+IPCShmReceive::~IPCShmReceive()
+{
+
+	shm_unlink(this->info.method_name);
+	if (pthread_cond_destroy(&(this->shm_ptr->cond_re)) != 0)
```

Comment:
`It shall be safe to destroy an initialized condition variable upon which no threads are currently blocked. Attempting to destroy a condition variable upon which other threads are currently blocked results in undefined behavior.`

https://linux.die.net/man/3/pthread_cond_destroy